native/src/main/java/io/ballerina/stdlib/task/TokenAcquisition.java

```diff
@@ -0,0 +1,189 @@
+
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.stdlib.task.utils.Utils;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+
+import static io.ballerina.stdlib.task.HealthCheckScheduler.startHealthCheckUpdater;
+
+public class TokenAcquisition {
+
+    public static final BString DB_HOST = StringUtils.fromString("host");
+    public static final BString DB_USER = StringUtils.fromString("user");
+    public static final BString DB_PASSWORD = StringUtils.fromString("password");
+    public static final BString DB_PORT = StringUtils.fromString("port");
+    public static final BString DATABASE = StringUtils.fromString("database");
+    public static final String JDBC_URL = "jdbc:mysql://%s:%d/%s";
+    public static final BString STATUS = StringUtils.fromString("status");
+    public static final BString ACTIVE_STATUS = StringUtils.fromString("active");
+    public static final BString STANDBY_STATUS = StringUtils.fromString("standby");
+    public static final BString INSTANCE_ID = StringUtils.fromString("instanceId");
+    public static final BString TOKEN_HOLDER = StringUtils.fromString("tokenholder");
+    public static final BString CONNECTION = StringUtils.fromString("connection");
+    public static final BString LIVENESS_INTERVAL = StringUtils.fromString("livenessInterval");
+    public static final String LAST_HEARTBEAT = "last_heartbeat";
+
+    public static class Error extends Exception {
+        public Error(String message) {
+            super(message);
+        }
+    }
+
+    public static BMap<BString, Object> acquireToken(BMap<Object, Object> databaseConfig,
+                                                     BString id, boolean tokenAcquired, int livenessInterval) {
+        DatabaseConfig dbConfig = new DatabaseConfig(
+                databaseConfig.getStringValue(DB_HOST).getValue(), databaseConfig.getStringValue(DB_USER).getValue(),
+                databaseConfig.getStringValue(DB_PASSWORD).getValue(), databaseConfig.getIntValue(DB_PORT).intValue(),
+                databaseConfig.getStringValue(DATABASE).getValue(), null
+        );
+        try {
+            String instanceId = id.getValue();
+            String jdbcUrl = String.format(JDBC_URL, dbConfig.host(), dbConfig.port(), dbConfig.database());
+            Connection connection = DriverManager.getConnection(jdbcUrl, dbConfig.user(), dbConfig.password());
+            connection.setAutoCommit(false);
+            tokenAcquired = attemptTokenAcquisition(connection, instanceId, tokenAcquired, livenessInterval);
+            connection.commit();
+            startHealthCheckUpdater(connection, instanceId, 1);
```

Comment:
Is it correct to call `startHealthCheckUpdater` always? Even if the token is not acquired. Don't we need to do a check whether `tokenAcquired` or not?